tests/integration/kubernetes/confidential/k8s-confidential.bats

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bats
+# Copyright 2022-2023 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+IMAGE_REPO="ghcr.io/confidential-containers/test-container"
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	( [ "${KATA_HYPERVISOR}" == "firecracker" ] || [ "${KATA_HYPERVISOR}" == "dragonball" ] || [ "${KATA_HYPERVISOR}" == "qemu-tdx" ] || [ "${KATA_HYPERVISOR}" == "qemu-snp" ] ) && skip "Test not supported yet."
+
+	pod_name="confidential-unencrypted-test"
+	container_name="confidential-unencrypted-container"
+	get_pod_config_dir
+
+	setup_unencrypted_confidential_pod
+}
+
+@test "Test unencrypted confidential container launch success and verify that we are running in a secure enclave." {
+	( [ "${KATA_HYPERVISOR}" == "firecracker" ] || [ "${KATA_HYPERVISOR}" == "dragonball" ] || [ "${KATA_HYPERVISOR}" == "qemu-tdx" ] || [ "${KATA_HYPERVISOR}" == "qemu-snp" ] ) && skip "Test not supported yet."
+	# Start the service/deployment/pod
+	kubectl apply -f "${TEST_DIR}/unencrypted-confidential.yaml"
+
+	# Retrieve pod name, wait for it to come up, retrieve pod ip
+	pod_name=$(kubectl get pod -o wide | grep ${pod_name} | awk '{print $1;}')
+
+	# Check pod creation
+	kubectl wait --for=condition=Ready --timeout=20 pod "$pod_name"
+
+	pod_ip=$(kubectl get pod -o wide | grep ${pod_name} | awk '{print $6;}')
+
+	# Look for SEV enabled in container dmesg output
+	coco_enabled=$(ssh -i ${SSH_KEY_FILE} \
+		-o "StrictHostKeyChecking no" \
+		-o "PasswordAuthentication=no" \
+		-t root@${pod_ip} \
+		'dmesg | grep ${DMESG_STRING}' || true)
```

Comment:
Fyi this will fail for now as regular Azure VM SKUs aren't confidential - but that capability will be available very soon. Will share more on this as soon as I know more.

Reply:
This will not run on reguer Azure VM SKUs, as, based on the ${KATA_HYPERVISOR}, those tests will only be running on machines we know for sure that are TEE capable.